internal/metadataproviders/system/metadata.go

```diff
@@ -159,3 +162,37 @@ func (p systemMetadataProvider) OSDescription(ctx context.Context) (string, erro
 func (systemMetadataProvider) HostArch() (string, error) {
 	return internal.GOARCHtoHostArch(runtime.GOARCH), nil
 }
+
+func (p systemMetadataProvider) HostIPs() (ips []net.IP, err error) {
+	ifaces, err := net.Interfaces()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, iface := range ifaces {
+		// skip if the interface is down or is a loopback interface
+		if iface.Flags&net.FlagUp == 0 || iface.Flags&net.FlagLoopback != 0 {
+			continue
+		}
+
+		addrs, errAddr := iface.Addrs()
+		if errAddr != nil {
+			return nil, fmt.Errorf("failed to get addresses for interface %v: %w", iface, errAddr)
+		}
+		for _, addr := range addrs {
+			ip, _, parseErr := net.ParseCIDR(addr.String())
+			if parseErr != nil {
+				return nil, fmt.Errorf("failed to parse address %q from interface %v: %w", addr, iface, parseErr)
+			}
+
+			if ip.IsLoopback() {
+				// skip loopback IPs
+				continue
+			}
```

Comment:
Is there a chance to have a loopback IP on an non loopback interface?

Reply:
I guess that is possible, but I couldn't find a way to test this. `IsLoopback` will test if the IP belongs to the `127.0.0.0/8` range (or the equivalent in IPv6), I think that is the best we can do.